src/software/utils/main_fisheyeProjection.cpp

```diff
@@ -0,0 +1,431 @@
+    ALICEVISION_COUT("Usage:\n\n" << allParams);
+    return EXIT_FAILURE;
+  }
+  catch(boost::program_options::error& e)
+  {
+    ALICEVISION_CERR("ERROR: " << e.what());
+    ALICEVISION_COUT("Usage:\n\n" << allParams);
+    return EXIT_FAILURE;
+  }
+
+  ALICEVISION_COUT("Program called with the following parameters:");
+  ALICEVISION_COUT(vm);
+
+  // set verbose level
+  system::Logger::get()->setLogLevel(verboseLevel);
+  
+  // check output folder and update to its absolute path
+  {
+    const fs::path outDir = fs::absolute(outputFolder);
+    outputFolder = outDir.string();
+  }
+
+  std::vector<std::string> imagePaths;
+  std::vector<float> times;
+  std::vector<oiio::ParamValueList> metadatas;
+
+  const boost::regex dateTimeExp(
+  // Syntax of Exif:DateTimeOriginal tag
+      "(?<date>[\\d\\:]*) "   // YYYY:MM:DD date separated with time with a space
+      "(?<h>\\d*):"  // HH: hour
+      "(?<m>\\d*):"  // MM: minutes
+      "(?<s>\\d*)"   // SS: seconds
+  );
+
+  for(const std::string& entry: inputPath)
+  {
+    const fs::path path = fs::absolute(entry);
+    if(fs::exists(path) && fs::is_directory(path))
+    {
+      for(fs::directory_entry& file : boost::make_iterator_range(fs::directory_iterator(path), {}))
+      {
+        if(image::isSupported(file.path().extension().string()))
+        {
+          imagePaths.push_back(file.path().string());
+
+          oiio::ParamValueList metadata = image::readImageMetadata(file.path().string());
+          metadatas.push_back(metadata);
+          std::string dateTime;
+          dateTime = metadata.get_string("Exif:DateTimeOriginal");
+
+          if(dateTime.empty())
+          {
+            ALICEVISION_LOG_ERROR("Can't sort images : no time metadata in " << file.path().string());
+            return EXIT_FAILURE;
+          }
+
+          // convert time from string to float to sort images by time for future stitching
+          boost::smatch what;
+          if(boost::regex_search(dateTime, what, dateTimeExp))
+          {
+              times.push_back(24.0 * std::stof(what["h"]) + 60.0 * std::stof(what["m"]) + std::stof(what["s"]));
+          }
+        }
+      }
+    }
+
+    else if(fs::exists(path) && image::isSupported(path.extension().string()))
+    {
+      imagePaths.push_back(path.string());
+
+      oiio::ParamValueList metadata = image::readImageMetadata(entry);
+      metadatas.push_back(metadata);
+      std::string dateTime;
+      dateTime = metadata.get_string("Exif:DateTimeOriginal");
+      if(dateTime.empty())
+      {
+        ALICEVISION_LOG_ERROR("Can't sort images : no time metadata in " << entry);
+        return EXIT_FAILURE;
+      }
+
+      // convert time from string to float to sort images by time for future stitching
+      boost::smatch what;
+      if(boost::regex_search(dateTime, what, dateTimeExp))
+      {
+          times.push_back(24.0 * std::stof(what["h"]) + 60.0 * std::stof(what["m"]) + std::stof(what["s"]));
```

Comment:
since this part is repeated here and in the previous if branch, maybe consider to refactor it into a function. the only difference is that here you are doing something with a single file, in the previous one you are doing the exact same thing for each file in the directory